qiime2/sdk/plugin_manager.py

```diff
@@ -62,22 +67,39 @@ def _init(self):
         for plugin in self.plugins.values():
             self._integrate_plugin(plugin)
 
+        for canonical_format in self._canonical_formats:
+            self._importable.update(
+                                self._reverse_transformers[canonical_format])
+            self._exportable.update(self.transformers[canonical_format])
+
+        for importable_format in self._importable:
+            if(isinstance(importable_format,
+                          qiime2.plugin.model.file_format._FileFormat)):
+                self._importable.add(self._ff_to_sfdf[importable_format])
+
+        for exportable_format in self._exportable:
+            if(isinstance(exportable_format,
+                          qiime2.plugin.model.file_format._FileFormat)):
+                self._exportable.add(self._ff_to_sfdf[exportable_format])
```

Comment:
My understanding is the three loops here are currently under revision, so I won't comment too much on them right now.